types/src/v0/impls/stake_table.rs

```diff
@@ -257,36 +304,39 @@ impl Membership<SeqTypes> for StaticCommittee {
             .filter(|entry| entry.stake() > U256::zero())
             .collect();
 
-        // Index the stake table by public key
-        let indexed_stake_table: BTreeMap<PubKey, _> = members
-            .iter()
-            .map(|entry| (PubKey::public_key(entry), entry.clone()))
-            .collect();
-
-        // Index the stake table by public key
-        let indexed_da_stake_table: BTreeMap<PubKey, _> = da_members
-            .iter()
-            .map(|entry| (PubKey::public_key(entry), entry.clone()))
-            .collect();
-
-        Self {
+        let committee = StaticCommittee {
             eligible_leaders,
             stake_table: HashSet::from_iter(members),
             da_stake_table: HashSet::from_iter(da_members),
-            indexed_stake_table,
-            indexed_da_stake_table,
+            // TODO: ??
+            indexed_stake_table: BTreeMap::new(),
+            indexed_da_stake_table: BTreeMap::new(),
             _epoch_size: 12, // TODO get the real number from config (I think)
-            _provider: L1Client::http(Url::from_str("http:://ab.b").unwrap()),
+            l1_client: L1Client::http(Url::from_str("http:://ab.b").unwrap()),
             _contract_address: None,
-        }
+        };
+
+        let committee = Arc::new(RwLock::new(committee));
+
+        let _handle = task::spawn(StaticCommittee::update_loop(committee.clone()));
```

Comment:
This is never spawned, since `new` is never called. See comment at the fn signature.

Reply:
Also I think the handle will get dropped and close, so we won't get any updates. We should hold on to the handle.

Reply:
Yeah, I just wanted to push the rough draft of what I plan to do. Also, I don't think dropping the handle will cancel the task. But yeah, we should hold on to the handle

Reply:
https://docs.rs/async-broadcast/latest/async_broadcast/struct.Receiver.html

Reply:
but we are not dropping the receiver? sorry if I am misunderstanding. Here the handle is dropped but the spawn task will continue to run and that task does not drop the receiver because of infinite loop.

Reply:
Oh yea sorry my mistake